src/script/calculator.ts

```diff
@@ -0,0 +1,488 @@
+import { saveHistory } from "./history";
+import { handleMC, handleMR, handleMS, handleMplusAndMinus } from "./memory";
+import { ExtendedMath } from "./utils"
+
+export class Calculator {
+  screen: HTMLElement;
+  calculationDone: boolean;
+  isDegreeMode: boolean;
+  isPrimary: boolean;
+  FEMode: boolean;
+  FEButton: HTMLElement;
+  sinBtn: HTMLElement;
+  cosBtn: HTMLElement;
+  tanBtn: HTMLElement;
+  isSecondPrimary: boolean=false;
+
+  private math: ExtendedMath;
+
+
+  constructor(screenId: string) {
+    this.screen = document.getElementById(screenId) as HTMLElement;
+    if (!this.screen) {
+      throw new Error(`Screen element with id "${screenId}" not found.`);
+    }
+    this.calculationDone = false;
+    this.screen.textContent = localStorage.getItem("calculationOutput") || "0";
+    this.isDegreeMode = true;
+    this.updateDegButton();
+    this.isPrimary = false;
+    this.FEMode = false;
+    this.handleKeyPress = this.handleKeyPress.bind(this);
+    this.math = Math as ExtendedMath;
+    
+
+    // Constants for buttons
+    const FEButton = document.getElementById("fe-btn") as HTMLElement;
+    const sinBtn = document.querySelector(".sin-btn") as HTMLElement;
+    const cosBtn = document.querySelector(".cos-btn") as HTMLElement;
+    const tanBtn = document.querySelector(".tan-btn") as HTMLElement;
+
+    if (!FEButton) {
+      console.error("F-E Button not found!");
+    }
+
+    this.FEButton = FEButton;
+    this.sinBtn = sinBtn;
+    this.cosBtn = cosBtn;
+    this.tanBtn = tanBtn;
+  }
+
+  public handleKeyPress(this: Calculator, event: KeyboardEvent): void {
+    const key: string = event.key;
```

Comment:
We can provide one early return if key is not found

```suggestion
    const key: string = event.key;
    
    if(!key) return;
```

Reply:
add https://github.com/Sakshi-Simform/calculator-ts/commit/b3685c79ca00f5bfb8e5bfae751b6118cd6a7522